src/Shared/FileUtilities.cs

```diff
@@ -421,6 +421,72 @@ internal static string FixFilePath(string path)
         }
 
         /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// Use a native shared method to massage file path. If the file is adjusted,
+        /// that qualifies is as a path.
+        /// </summary>
+        internal static string MaybeAdjustFilePath(string value)
+        {
+            if (NativeMethodsShared.IsWindows || string.IsNullOrWhiteSpace(value) ||
+                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\"))
+            {
+                return value;
+            }
+
+            // For Unix-like systems, we may want to convert backslashes to slashes
+            string newValue = Regex.Replace(value, @"[\\/]+", "/");
```

Comment:
FWIW, time spent inside the MSBuild engine is usually a pretty small proportion of the overall build time, so on a one-minute build I would expect even a real regression not to show up as any sort of human-visible difference.  (Unless the regression was excessively large, of course.  Or unless you put together a fake project that did nothing but exercise the changed codepath.) 

Whether or not it's worth digging deeper into whether / how much this codepath has regressed probably depends on what the performance bar is.